[dim_order] raised runtime error when tensor has ambiguous dim order

Gasoonjia opened the pull request:
Summary: This diff makes tensor.dim_order() raise error when tensor's dim order is ambiguous. Detail discussion can be found https://fb.workplace.com/groups/894363187646754/permalink/2039987243084337/

Test Plan: buck2 test fbcode//mode/dev-nosan fbcode//caffe2/test:torch -- test_dim_order

Differential Revision: D65133579

larryliu0820 reviewed (approved):
Looks like @ezyang and @digantdesai 's comments are addressed. Accepting. We can always improve in follow up PRs.

Gasoonjia commented:
@pytorchbot label "release notes: nn"

Gasoonjia committed:
[dim_order] raised runtime error when tensor has ambiguous dim order (#139180)

Summary:

This diff makes tensor.dim_order() raise error when tensor's dim order is ambiguous. Detail discussion can be found https://fb.workplace.com/groups/894363187646754/permalink/2039987243084337/

Test Plan:
```
 buck2 test fbcode//mode/dev-nosan fbcode//caffe2/test:torch -- test_dim_order
```

Reviewed By: larryliu0820

Differential Revision: D65133579

Gasoonjia commented:
Abandoned. Working on https://github.com/pytorch/pytorch/pull/141632 instead.
